docs/pages/developers/evm/intent-gateway-sdk.mdx

```diff
@@ -0,0 +1,783 @@
+
+## Canceling Orders
+
+Order cancellation proves on-chain that an order was not filled before its deadline. The SDK automates fetching proofs, tracking finality, submitting the Hyperbridge message, and resuming from interruptions.
+
+### When to Cancel
+
+You can cancel an order when:
+
+- The deadline has passed (order expired)
+- The order has not been filled, which can be due to calldata execution reverting.
+- You want to recover escrowed tokens and fees
+
+### The Cancellation Process
+
+The SDK orchestrates the [ISMP GET request](/protocol/ismp/requests) flow. Your responsibilities are:
+
+1. Start the cancellation generator.
+2. Wait for `DESTINATION_FINALIZED` and read the finalized height.
+3. Submit the `cancelOrder` transaction on source chain with `{ relayerFee, height }`.
+4. Provide the resulting transaction hash back to the generator when it yields `AWAITING_GET_REQUEST`.
+5. Iterate until `HYPERBRIDGE_FINALIZED`.
+
+### Basic Cancellation
+
+```typescript
+import { IndexerClient } from "@hyperbridge/sdk/client";
+import { createQueryClient } from "@hyperbridge/sdk/query-client";
+import { EvmChain, SubstrateChain } from "@hyperbridge/sdk/chain";
+
+// 1. Create/reuse the indexer client
+const queryClient = createQueryClient({
+  url: "https://indexer.hyperbridge.network",
+});
+const sourceChain = new EvmChain({
+  chainId: 56,
+  host: bscHostAddress,
+  rpcUrl: "https://bsc-dataseed.binance.org",
+});
+const destChain = new EvmChain({
+  chainId: 1,
+  host: ethHostAddress,
+  rpcUrl: "https://eth-mainnet.g.alchemy.com/v2/YOUR_API_KEY",
+});
+const hyperbridgeChain = new SubstrateChain({
+  wsUrl: "wss://hyperbridge.polkadot.io",
+  stateMachineId: "KUSAMA-4009",
+  consensusStateId: "PAS0",
+  hasher: "Keccak",
+});
+await hyperbridgeChain.connect();
+
+const indexerClient = new IndexerClient({
+  source: sourceChain,
+  dest: destChain,
+  hyperbridge: hyperbridgeChain,
+  queryClient,
+  pollInterval: 1_000,
+});
+
+// 2. Start cancellation (generator)
+const cancelGen = intentGateway.cancelOrder(order, indexerClient);
+
+// 3. Wait for destination finalization
+let step = await cancelGen.next();
+while (!step.done && step.value?.status !== "DESTINATION_FINALIZED") {
+  step = await cancelGen.next();
+}
+
+const finalizedHeight = step.value.data.proof.height;
+```
+
+### Submit cancel transaction and continue
+
+```typescript
+import { parseUnits } from "viem";
+
+// 4. Submit cancel transaction on the source chain
+const cancelOptions = {
+  relayerFee: parseUnits("0.05", 18),
+  height: finalizedHeight,
+};
+
+// Resolve the IntentGateway address on the source chain
+const intentGatewayAddress = sourceChain.configService.getIntentGatewayAddress(
+  sourceChain.config.stateMachineId
+);
+
+const hash = await walletClient.writeContract({
+  address: intentGatewayAddress,
+  abi: IntentGatewayABI.ABI,
+  functionName: "cancelOrder",
+  args: [order, cancelOptions],
```

Comment:
cancelOrder needs a value parameter that should be equivalent to the protocol fees for dispatching the get request

Reply:
And where to get the value?

Reply:
An approximate value is okay?